libmachine/provision/utils.go

```diff
@@ -166,16 +173,12 @@ func ConfigureAuth(p Provisioner, authOptions auth.AuthOptions) error {
 		return err
 	}
 
-	return nil
-}
+	// TODO: Do not hardcode daemon port, ask the driver
+	if err := utils.WaitForDocker(ip, dockerPort); err != nil {
```

Comment:
This fixes a race which was causing issues in the integration tests - basically, there is a small window after a create finishes where the docker daemon is still starting up.  It's really hard to notice if doing things by hand but if you want to run a `docker $(docker-machine config host) ps` type command _immediately_ after `docker-machine create` you're gonna have a bad time without this.